lib/demo_web/controllers/user_auth.ex

```diff
@@ -0,0 +1,116 @@
+defmodule DemoWeb.UserAuth do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  alias Demo.Accounts
+  alias DemoWeb.Router.Helpers, as: Routes
+
+  # Make the remember me cookie valid for 60 days.
+  # If you want bump or reduce this value, also change
+  # the token expiry itself in UserToken.
+  @max_age 60 * 60 * 24 * 60
+  @remember_me_cookie "user_remember_me"
+
+  @doc """
+  Logs the user in.
+
+  It deletes the CSRF token and renews the session
+  to avoid fixation attacks.
+  """
+  def login_user(conn, user, params \\ %{}) do
+    Plug.CSRFProtection.delete_csrf_token()
+    token = Accounts.generate_session_token(user)
+
+    user_return_to = get_session(conn, :user_return_to)
+    delete_session(conn, :user_return_to)
+
+    conn
+    |> put_session(:user_token, token)
+    |> configure_session(renew: true)
+    |> maybe_write_remember_me_cookie(token, params)
+    |> redirect(to: user_return_to || signed_in_path(conn))
+  end
+
+  defp maybe_write_remember_me_cookie(conn, token, %{"remember_me" => "true"}) do
+    put_resp_cookie(conn, @remember_me_cookie, token, sign: true, max_age: @max_age)
```

Comment:
Would be nice if this could use the session cookie opts for attrtibutes like `secure` or `SameSite`.

Reply:
Ah, good point. Unfortunately there is no way to retrieve the session options downstream. We would need to add that first. 🤔 At least `secure` should be automatically set whenever using `https` but or `SameSite` you are right. I will at least move these options to the top of the module to make it clearer.